go/arrow/ipc/file_reader.go

```diff
@@ -476,6 +487,9 @@ func (ctx *arrayLoaderContext) loadArray(dt arrow.DataType) arrow.ArrayData {
 	case *arrow.BinaryType, *arrow.StringType, *arrow.LargeStringType, *arrow.LargeBinaryType:
 		return ctx.loadBinary(dt)
 
+	case arrow.BinaryViewDataType:
```

Comment:
`BinaryViewDataType` is an interface, not a concrete type. `StringViewDataType` implements the `BinaryViewDataType` interface. The concrete types are `BinaryViewType` and `StringViewType`. The types that end with `DataType` in the name are interfaces (like `FixedWidthDataType`). So we don't need a separate case for `StringViewType` because they are processed identically